src/optimize.d

```diff
@@ -796,8 +796,23 @@ extern (C++) Expression Expression_optimize(Expression e, int result, bool keepL
                 return;
             }
             // If e2 *could* have been an integer, make it one.
-            if (e.e2.op == TOKfloat64 && (e.e2.toReal() == cast(sinteger_t)e.e2.toReal()))
-                e.e2 = new IntegerExp(e.loc, e.e2.toInteger(), Type.tint64);
+            if (e.e2.op == TOKfloat64)
+            {
+                version (all)
```

Comment:
I don't like adding dead code, and I've put quite a lot of effort into removing it.  I have no problem with the change, and I've done similar things myself, my problem is with the dead code Walter asked for.

If we're going to see dozens of similar changes then it might be better to look at other options (like patching the binaries to replace the prefix with a nop).